Src/HALAL/Services/Communication/Ethernet/TCP/Socket.cpp

```diff
@@ -201,5 +243,23 @@ void Socket::error_callback(void *arg, err_t error){
 	ErrorHandler("Client socket error: %d. Socket closed",error);
 }
 
+void Socket::connection_error_callback(void *arg, err_t error){
```

Comment:
maybe I am missing something but you pass a **void*** and instantly cast it to a **Socket*** by copying it.
also I assume it cannot be null, since you don't check for it. If that's the case just pass a **Socket&**, we avoid the copy and also avoid passing a **void***

Reply:
the lwip library needs to call this function in order for it to work, and the way its made is that the function has to be passed with a setter function (in this case tcp_err) after you construct the pcb. 

The function the tcp_err receives as a parameter has to be specifically a null pointer + an err_t value, in order for it to work. Passing any other type of function would lead into an invalid conversion error.

Reply:
i closed the other change request so we can talk it in this one. If you have an idea to get around the lwip type problem put it here. If you are fine with the explanation you are free to close this one too.